design/dsl/endpoint.go

```diff
@@ -32,7 +32,7 @@ func Endpoint(name string, dsl func()) {
 		eval.IncompatibleDSL()
 		return
 	}
-	ep := &design.EndpointExpr{Name: name, DSLFunc: dsl}
+	ep := &design.EndpointExpr{Name: name, Service: s, DSLFunc: dsl}
```

Comment:
@raphael wasn't sure about this change but it seemed needed. Let me know if it is wrong